src/core/util/props.js

```diff
@@ -94,6 +119,7 @@ function assertProp (
   vm: ?Component,
   absent: boolean
 ) {
+  assertPropObject(prop, name, vm)
```

Comment:
Here you need to add production environment check so this function can be removed by minifier, [like here](https://github.com/Alex-Sokolov/vue/blob/bb8a124214b2d2d19162f65e903b6e42df57e0dc/src/core/util/props.js#L55)